text/0007-repo.md

```diff
@@ -0,0 +1,81 @@
+- Feature Name: repo
+- Start Date: 
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+repo defines an interface for local storage & retrieval of qri data.
+
+# Motivation
+[motivation]: #motivation
+
+TODO
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+TODO
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Repo is the interface for working with a qri repository qri repos are stored
+graph of resources:datasets, known peers, analytics data, change requests, etc.
+Repos are connected to a single peer profile. Repos must wrap an underlying 
+cafs.Filestore, which is intended to act as the canonical store of state across 
+all peers that this repo may interact with.
+
+```golang
+type Repo interface {
+  // All repositories wraps a content-addressed filestore as the cannonical
+  // record of this repository's data. Store gives direct access to the
+  // cafs.Filestore instance any given repo is using.
+  Store() cafs.Filestore
+
+  // All Repos must keep a Refstore, defining a store of known datasets
+  Refstore
+  // EventLog keeps a log of Profile activity for this repo
+  EventLog
+
+  // A repository must maintain profile information about the owner of this dataset.
+  // The value returned by Profile() should represent the peer.
+  Profile() (*profile.Profile, error)
+  // SetProfile sets this repo's current profile. Profiles must contain a private key
+  SetProfile(*profile.Profile) error
+  // PrivateKey hands over this repo's private key
+  PrivateKey() crypto.PrivKey
+  // A repository must maintain profile information about encountered peers.
+  // Decsisions regarding retentaion of peers is left to the the implementation
```

Comment:
"Decisions"

If we can change this interface, I am not at all a fan of having two methods with completely different semantics that only vary in name by a single letter. Profile represents the peer's own info, while Profiles is other peers we've seen in the wild, which is different in more important ways than simply being plural. I would rename it to something like "RemoteProfiles" or "KnownProfiles".

Reply:
Throwing "PeerProfiles" into the ring.